src/main/scala/is/hail/methods/Skat.scala

```diff
@@ -0,0 +1,287 @@
+package is.hail.methods
+
+import is.hail.utils._
+import is.hail.variant._
+import is.hail.expr._
+import is.hail.keytable.KeyTable
+import is.hail.stats.{ RegressionUtils, SkatModel }
+import is.hail.annotations.Annotation
+import breeze.linalg._
+import org.apache.spark.sql.Row
+
+object SkatStat {
+  val schema = TStruct(
+    ("q", TFloat64),
+    ("pval", TFloat64),
+    ("fault", TInt32))
+}
+
+case class SkatStat(q: Double, pval: Double, fault: Int) {
+  def toAnnotation = Annotation(q, pval,fault)
+}
+
+case class SkatTuple[T <: Vector[Double]](q: Double, xw: T, qtxw: DenseVector[Double])
+
+object SkatAgg {
+  val zeroValDense  = SkatAgg(0.0, new ArrayBuilder[DenseVector[Double]](), new ArrayBuilder[DenseVector[Double]]())
```

Comment:
Using an array builder in this way feels wrong. I think it might be better to `groupByKey` instead of doing this aggregate business